### PR TITLE
[storybook__addon-storyshots-puppeteer] new types definition

### PR DESCRIPTION
**Pull request by Yama-Tomo**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If removing a declaration:
- [x] If a package was never on DefinitelyTyped, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

File: types/storybook__addon-storyshots/index.d.ts [hunk 1]
```diff
@@ -1,6 +1,7 @@
-// Type definitions for @storybook/addon-storyshots 4.0
+// Type definitions for @storybook/addon-storyshots 5.1
```
Comment (jessetrinity) [inline types/storybook__addon-storyshots/index.d.ts]: Can you [move this to an old version](https://github.com/DefinitelyTyped/DefinitelyTyped#if-a-library-is-updated-to-a-new-major-version-with-breaking-changes-how-should-i-update-its-type-declaration-package) instead?

**Review by jessetrinity — CHANGES_REQUESTED** [review]

File: types/storybook__addon-storyshots/index.d.ts (hunk 1 above)
Comment (Yama-Tomo) [reply types/storybook__addon-storyshots/index.d.ts]: @jessetrinity Thank you for your review. But `imageSnapshot` was already removed in v4.0.0 this package. Should I still do it?

detail: https://github.com/storybookjs/storybook/releases/tag/v4.0.0-alpha.10
> `Extract imageSnapshots to a separate package #3742`

Comment (jessetrinity) [reply types/storybook__addon-storyshots/index.d.ts]: Ah. Well if the types for 4.0 are complete and won't need to be updated and there are not breaking changes it should be fine.

Comment (Yama-Tomo) [reply types/storybook__addon-storyshots/index.d.ts]: ok, please approve if there are no other problems 👍

**Pull request merged**